lib/lf/fe/mesh_function_global.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file
+ * @brief Transforms an function expressed in global coordinates into a
+ * MeshFunction.
+ * @author Raffael Casagrande
+ * @date   2018-12-15 09:29:05
+ * @copyright MIT License
+ */
+
+#ifndef __1323b89d2f584608940fcb37657142cc
+#define __1323b89d2f584608940fcb37657142cc
+#include <vector>
+
+#include <lf/mesh/mesh.h>
+
+namespace lf::fe {
+
+template <class F>
+class MeshFunctionGlobal {
+  using F_return_type =
+      decltype(std::declval<F>()(std::declval<Eigen::Vector2d>()));
+
+ public:
+  MeshFunctionGlobal(const MeshFunctionGlobal&) = default;
+  MeshFunctionGlobal(MeshFunctionGlobal&&) = default;
+  MeshFunctionGlobal& operator=(const MeshFunctionGlobal&) = delete;
+  MeshFunctionGlobal& operator=(MeshFunctionGlobal&&) = delete;
+
+  MeshFunctionGlobal(const F& f) : f_(f) {}
```

Comment:
Does this work with lambda functions defined right inside the function call.

Reply:
Yes this works with lambda functions

Reply:
Then we should use this passing by reference for all functor arguments. It mainly concerns classes in the 'fe' namespace.